src/core/utils/misc.js

```diff
@@ -54,3 +54,10 @@ export function forEachAsync(array, func) {
     });
   });
 }
+
+export function useIfDefined(value, defaultValue) {
+  if (value !== undefined) {
```

Comment:
This should use `if (typeof value !== 'undefined'`). `undefined` can be redefined in the global namespace in JavaScript.

Reply:
I'm not sure this is allowed nowadays. And if anyone is doing that, they have bigger problems than this check not working out ;) If there are any other changes required, I can make this one. But it's not worth the hassle on its own, imo.

Reply:
Ok so it seems that `undefined` is no longer writable in the global namespace in JavaScript since 1.8.5 according to MDN. However, this is definitely possible nowadays using a self executing function.  Consider the two following code snippets.

```javascript
(function() { 
  var undefined = 'foo'; 
  console.log(undefined, typeof undefined);
})();
```

```javascript
(function(undefined) { 
  console.log(undefined, typeof undefined); 
})('foo');
```

Both code snippets will print `foo string` to the console. For this reason we should use the `typeof` way of checking if a value is `undefined`.

Reply:
Yes, it is possible to name a variable "undefined". I honestly think the chances of this happening here are practically 0. But I made the change and will have Tsvetomir start over with testing tomorrow.